qutebrowser/mainwindow/tabbedbrowser.py

```diff
@@ -930,3 +936,11 @@ def callback(ok):
             tab.scroller.to_point(point)
         else:
             message.error("Mark {} is not set".format(key))
+
+    def save_undo_stack(self):
+        """Return the stack of UndoEntries."""
+        return self._undo_stack.copy()
+
+    def restore_undo_stack(self, stack):
+        """Set the stack of UndoEntries."""
+        self._undo_stack = stack.copy()
```

Comment:
Do you think it makes sense to do those copies? We only save the undo stack before the window is destroyed anyways (so it shouldn't ever get mutated) and when we restore it, we don't need the passed stack anymore - so I don't think those copies are needed?